### PR TITLE
Create reference data generic import/export

### PR DESCRIPTION
**Pull request by fitn**
This PR introduces:
- Generic import for reference data
- Generic export for reference data
- Rework on reference data API (saver, remover, updater)

### CODE REVIEW

File: Processor/Denormalization/Processor.php [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+use Akeneo\Component\StorageUtils\Detacher\ObjectDetacherInterface;
+use Doctrine\Common\Util\Inflector;
+use Doctrine\ORM\EntityManagerInterface;
+use Pim\Bundle\CustomEntityBundle\Configuration\Registry;
+use Pim\Bundle\CustomEntityBundle\Entity\Repository\CustomEntityRepository;
+use Pim\Component\Connector\Exception\InvalidItemFromViolationsException;
+use Symfony\Component\Validator\ConstraintViolationListInterface;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+
+/**
```
Comment (jmleroux) [inline Processor/Denormalization/Processor.php]: here a little description of what this processor does.

File: Resources/config/jobs_import.yml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        tags:
+            - { name: pim_import_export.job_parameters.form_configuration_provider }
+
+    pim_custom_entity.job.job_parameters.default_values_provider.simple_csv_import:
+        class: '%pim_connector.job.job_parameters.default_values_provider.simple_csv_import.class%'
+        arguments:
+            -
+                - '%pim_custom_entity.job.import.custom_entity.name%'
+        tags:
+            - { name: akeneo_batch.job.job_parameters.default_values_provider }
```
Comment (jmleroux) [inline Resources/config/jobs_import.yml]: this file is quite big. What about splitting it?

File: Connector/Job/JobParameters/ConstraintCollectionProvider/ReferenceData.php [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+use Akeneo\Component\Batch\Job\JobInterface;
+use Akeneo\Component\Batch\Job\JobParameters\ConstraintCollectionProviderInterface;
+use Pim\Bundle\CustomEntityBundle\Configuration\Registry;
+use Symfony\Component\Validator\Constraints\Choice;
+use Symfony\Component\Validator\Constraints\Collection;
+use Symfony\Component\Validator\Constraints\NotBlank;
+
+/**
+ * @author Romain Monceau <romain@akeneo.com>
```
Comment (fitn) [inline Connector/Job/JobParameters/ConstraintCollectionProvider/ReferenceData.php]: Add phpdoc + license

File: Connector/Job/JobParameters/DefaultValuesProvider/ReferenceData.php [hunk 4]
```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Pim\Bundle\CustomEntityBundle\Connector\Job\JobParameters\DefaultValuesProvider;
+
+use Akeneo\Component\Batch\Job\JobInterface;
+use Akeneo\Component\Batch\Job\JobParameters\DefaultValuesProviderInterface;
+
+/**
+ * @author Romain Monceau <romain@akeneo.com>
```
Comment (fitn) [inline Connector/Job/JobParameters/DefaultValuesProvider/ReferenceData.php]: phpdoc + license

File: Connector/Job/JobParameters/FormConfigurationProvider/ReferenceData.php [hunk 5]
```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Pim\Bundle\CustomEntityBundle\Connector\Job\JobParameters\FormConfigurationProvider;
+
+use Akeneo\Component\Batch\Job\JobInterface;
+use Pim\Bundle\CustomEntityBundle\Configuration\Registry;
+use Pim\Bundle\ImportExportBundle\JobParameters\FormConfigurationProviderInterface;
+
+/**
+ * @author Romain Monceau <romain@akeneo.com>
```
Comment (fitn) [inline Connector/Job/JobParameters/FormConfigurationProvider/ReferenceData.php]: phpdoc + license

File: Connector/Processor/Denormalization/Processor.php [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+use Akeneo\Component\StorageUtils\Updater\ObjectUpdaterInterface;
+use Doctrine\ORM\EntityManagerInterface;
+use Pim\Bundle\CustomEntityBundle\Configuration\Registry;
+use Pim\Bundle\CustomEntityBundle\Entity\Repository\CustomEntityRepository;
+use Pim\Component\Connector\Exception\InvalidItemFromViolationsException;
+use Symfony\Component\Validator\ConstraintViolationListInterface;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+
+/**
+ * @author Romain Monceau <romain@akeneo.com>
```
Comment (fitn) [inline Connector/Processor/Denormalization/Processor.php]: phpdoc + license

File: Connector/Processor/Denormalization/Processor.php [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+    /**
+     * @return CustomEntityRepository
+     */
+    protected function getRepository()
+    {
+        return $this->em->getRepository($this->getClassName());
+    }
+
+    /**
+     * Gets class name from the conf registry
```
Comment (fitn) [inline Connector/Processor/Denormalization/Processor.php]: Gets reference data class name from the conf registry

File: Connector/Processor/Normalization/Processor.php [hunk 8]
```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace Pim\Bundle\CustomEntityBundle\Connector\Processor\Normalization;
+
+use Akeneo\Component\Batch\Item\ItemProcessorInterface;
+use Symfony\Component\PropertyAccess\PropertyAccessorInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+
+/**
+ * @author Romain Monceau <romain@akeneo.com>
```
Comment (fitn) [inline Connector/Processor/Normalization/Processor.php]: phpdoc + license

File: Connector/Processor/Normalization/Processor.php [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+            $value = $this->propertyAccessor->getValue($item, $property->getName());
+            $normalizedData[$property->getName()] = $this->normalizer->normalize($value, 'flat');
+        }
+
+        return $normalizedData;
+    }
+
+    /**
+     * @param array $skippedFields
```
Comment (fitn) [inline Connector/Processor/Normalization/Processor.php]: phpdoc to explain that you can inject specific fields

File: Connector/Reader/Database/ReferenceDataReader.php [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+
+namespace Pim\Bundle\CustomEntityBundle\Connector\Reader\Database;
+
+use Akeneo\Component\Batch\Model\StepExecution;
+use Doctrine\ORM\EntityManagerInterface;
+use Pim\Bundle\CustomEntityBundle\Configuration\Registry;
+use Pim\Component\Connector\Reader\Database\AbstractReader;
+
+/**
+ * @author Romain Monceau <romain@akeneo.com>
```
Comment (fitn) [inline Connector/Reader/Database/ReferenceDataReader.php]: phpdoc + license

File: Manager/Manager.php [hunk 11]
```diff
@@ -15,78 +16,68 @@
...
+    public function __construct(
+        EntityManagerInterface $em,
+        ObjectUpdaterInterface $updater,
+        SaverInterface $saver,
+        RemoverInterface $remover
+    ) {
+        $this->em      = $em;
+        $this->updater = $updater;
+        $this->saver   = $saver;
+        $this->remover = $remover;
```
Comment (fitn) [inline Manager/Manager.php]: BC breaks to add in CHANGELOG.md

File: Updater/Updater.php [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+namespace Pim\Bundle\CustomEntityBundle\Updater;
+
+use Akeneo\Component\StorageUtils\Updater\ObjectUpdaterInterface;
+use Doctrine\Common\Util\ClassUtils;
+use Doctrine\Common\Util\Inflector;
+use Pim\Component\ReferenceData\Model\ReferenceDataInterface;
+use Symfony\Component\PropertyAccess\PropertyAccessorInterface;
+
+/**
+ * @author Romain Monceau <romain@akeneo.com>
```
Comment (fitn) [inline Updater/Updater.php]: phpdoc + license

**Pull request merged**